docs/decisions/0014-single-taxonomy-view-api.rst

```diff
@@ -0,0 +1,126 @@
+13. Single taxonomy view API
+=====================================
+
+
+Context
+--------
+
+This view returns tags of a closed taxonomy (for MVP has not been implemented yet
+for open taxonomies). It is necessary to make a decision about what structure the tags are going 
+to have and how the pagination is going to work. It was taken into account that a taxonomy can
+have a large number of tags and are mostly represented as trees.
+
+**Branch:** Is the representation of a root Tag and all its children up to the leaves.
+
+
+Decision
+---------
+
+
+Tag representation
+~~~~~~~~
+
+Return a list of root tags and within each one have the list of children tags. Each root would have
+its entire branch. The list of root tags will be ordered alphabetically as is each listing at each level of the tree.
+
+{
+    "count": 100,
+    "tags": [
+        {
+            "id": "tag_1",
+            "value": "Tag 1",
+            "taxonomy_id": "1",
+            "sub_tags": [
+                {
+                    "id": "tag_2",
+                    "value": "Tag 2",
+                    "taxonomy_id": "1",
+                    "sub_tags": [
+                        (....)
+                    ]
+                },
+                (....)
+            ]
+        },
+        (....)
+    ]
+}
+
+
+**Pros:**
+
+- The edX's interfaces show the tags in the form of a tree.
+- The frontend needs no further processing as it is in a displayable format.
+- It is kept as a simple implementation.
+
+**Cons:**
+
+- More implementation on the API side.
+
+
+Pagination
+~~~~~~~~~~~
+
+Apply the pagination only in the root tags and bring the entire branch of each root.
+Children do not affect pagination in any way.
+
+**Pros**
+
+- It is the simplest way.
+
+**Cons**
+
+- The children would not have pagination, in the long run there may be cases in which
+  the branch has hundreds of children, and they would still all be brought.
```

Comment:
If we need to mitigate this, we could enforce a "maximum number of tags per branch" during tag import.

Reply:
We can ask the product folks, but I don't think this restriction is going to work. This whole approach assumes that there are a lot of root tags and very few children per root, but I think the opposite will generally be more common: only a few root tags and potentially thousands of child or grandchild tags. Think of the tree of life used in Biology - it has only three root tags (Bacteria, Achaea, and Eukaryota) but there are over 1 million species in the taxonomy. Of course that's an extreme case, but I think that shape of tree is more common than many root tags and few children. See also LabXchange, or Amazon product categories, or anything like that.

Reply:
Oh Yes, you are right, it makes sense. I think we can mitigate the "_Cons_" of "Get the branch in another call"